src/main/java/io/aiven/connect/elasticsearch/clientwrapper/AivenElasticsearchClientWrapper.java

```diff
@@ -0,0 +1,376 @@
+                "Couldn't start ElasticsearchSinkTask due to connection error:",
+                e
+            );
+        }
+    }
+
+    // visible for testing
+    public AivenElasticsearchClientWrapper(final String address) {
+        try {
+            final Map<String, String> props = new HashMap<>();
+            props.put(ElasticsearchSinkConnectorConfig.CONNECTION_URL_CONFIG, address);
+            props.put(ElasticsearchSinkConnectorConfig.TYPE_NAME_CONFIG, "kafka-connect");
+            this.elasticClient = getElasticsearchClient(new ElasticsearchSinkConnectorConfig(props));
+            this.version = getServerVersion();
+        } catch (final IOException e) {
+            throw new ConnectException(
+                "Couldn't start ElasticsearchSinkTask due to connection error:",
+                e
+            );
+        } catch (final ConfigException e) {
+            throw new ConnectException(
+                "Couldn't start ElasticsearchSinkTask due to configuration error:",
+                e
+            );
+        }
+    }
+
+    // visible for testing
+    public AivenElasticsearchClientWrapper(final ElasticsearchSinkConnectorConfig config) {
+        try {
+            this.elasticClient = getElasticsearchClient(config);
+            this.version = getServerVersion();
+        } catch (final IOException e) {
+            throw new ConnectException(
+                "Couldn't start ElasticsearchSinkTask due to connection error:",
+                e
+            );
+        } catch (final ConfigException e) {
+            throw new ConnectException(
+                "Couldn't start ElasticsearchSinkTask due to configuration error:",
+                e
+            );
+        }
+    }
+
+    private Version getServerVersion() throws IOException {
+        final String esVersion = this.elasticClient.info(RequestOptions.DEFAULT).getVersion().getNumber();
+        return matchVersionString(esVersion);
+    }
+
+    private Version matchVersionString(final String esVersion) {
+        // Default to newest version for forward compatibility
+        final Version defaultVersion = Version.ES_V8;
+        if (esVersion == null) {
+            LOG.warn("Couldn't get Elasticsearch version, version is null");
+            return defaultVersion;
+        } else if (esVersion.startsWith("1.")) {
+            return Version.ES_V1;
+        } else if (esVersion.startsWith("2.")) {
+            return Version.ES_V2;
+        } else if (esVersion.startsWith("5.")) {
+            return Version.ES_V5;
+        } else if (esVersion.startsWith("6.")) {
+            return Version.ES_V6;
+        } else if (esVersion.startsWith("7.")) {
+            return Version.ES_V7;
+        } else if (esVersion.startsWith("8.")) {
+            return Version.ES_V8;
+        }
+        return defaultVersion;
+    }
+
+    private boolean es8compat(final Version version) {
+        return Objects.requireNonNull(version) == Version.ES_V8;
```

Comment:
not blocking, just leaving as note that we should reason later. Probably here it should be a `>=` instead of an `==`, the break in the compatibility should be mantained from now on